### ISSUE TITLE
FX115 ( Blends) Missing pixels on peek and fixtures 

### ISSUE THREAD

**Issue by dosipod** [Aircoookie/WLED#4335]
### What happened?

FX115 ( Blends) missing pixels 
![image](https://github.com/user-attachments/assets/30f42792-14b4-4b43-8e46-e309f0cae80b)

### To Reproduce Bug

Use FX115 ( Blends)  and you will notice missing pixels on peek and fixtures 

### Expected Behavior

No missing pixels  

### Install Method

Self-Compiled

### What version of WLED?

0.15.0-rc1 (build 2411250)

### Which microcontroller/board are you seeing the problem on?

ESP32

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blazoncek** [comment]
FYI Blends is a 1D effect with limitations to accommodate ESP8266.

**Comment by softhack007** [comment]
Looks like the effect should be "modernised" ... it creates a fastLed - style pixel buffer the effect cannot handle more than 255 pixels in total.

I think the complete "pixels" buffer is not needed and it could be replaced by `getPixelColor`.

This loop could use getPixelColor instead of the pixels buffer
https://github.com/Aircoookie/WLED/blob/a121f5b61b2dbb2934630120f22527a5cdab91d4/wled00/FX.cpp#L4492-L4495

The second loop just writes the pixels back
https://github.com/Aircoookie/WLED/blob/a121f5b61b2dbb2934630120f22527a5cdab91d4/wled00/FX.cpp#L4498-L4501

**Comment by dosipod** [comment]
It is fixed now on MM

**Comment by dosipod** [comment]
The two 16x16 and 32x16 fixtures I tested with in the beginning  of the post are okay now  without any missing pixels on effect default .

Below I would not call it a bug because ARC might be meant to behave this way so just to log test results with few scenarios ( I even like the last test with 32x16 having a hole in the middle ) 
![image](https://github.com/user-attachments/assets/7511fbda-c52a-4d8c-8bbf-267e3d2999a5)

You may close at your discretion

**Comment by blazoncek** [comment]
@dosipod that's a problem Arc expansion has on some dimensions of segments. But may be related to how that particular effect calculates its maximum dimension.

**Comment by dosipod** [comment]
@blazoncek Yeah I see the same with multiple effects when using Arc so not an issue with Blends , as long as there is symmetry it is fine

**Comment by DedeHai** [comment]
> i would you like to keep this report open, as a reminder for modernizing the logic?

does it need modernizing? it will change the looks if you remove the buffer (on all setups that have global buffer disabled), not sure its worth it.
